lib/core-integration/src/Test/Integration/Scenario/API/Shelley/Transactions.hs

```diff
@@ -656,6 +660,102 @@ spec = describe "SHELLEY_TRANSACTIONS" $ do
                 (#balance . #available)
                 (`shouldBe` Quantity (faucetAmt - feeEstMax - amt)) ra2
 
+    it "TRANS_CREATE_10 - Multi-asset balance" $ \ctx -> runResourceT $ do
+        w <- fixtureMultiAssetWallet ctx
+        r <- request @ApiWallet ctx (Link.getWallet @'Shelley w) Default Empty
+        verify r
+            [ expectField (#assets . #available) (`shouldNotBe` mempty)
+            , expectField (#assets . #total) (`shouldNotBe` mempty)
+            ]
+
+    it "TRANS_CREATE_10 - Multi-asset transaction with Ada" $ \ctx -> runResourceT $ do
+
+        wSrc <- fixtureMultiAssetWallet ctx
+        wDest <- emptyWallet ctx
+        ra <- request @ApiWallet ctx (Link.getWallet @'Shelley wSrc) Default Empty
+        let (_, Right wal) = ra
+
+        -- pick out an asset to send
+        let assetsSrc = wal ^. #assets . #total . #getApiT
+        assetsSrc `shouldNotBe` mempty
+        let TokenBundle.AssetId pid an = Set.elemAt 0 (TokenMap.getAssets assetsSrc)
+        let val = [(toText pid, toText an, minUTxOValue)]
+
+        -- Allow for a higher minimum utxo coin due to assets
+        let minAda = minUTxOValue * 2
```

Comment:
I think we should ultimately provide this as an API endpoint.

Reply:
Sounds good.